src/main/java/lotto/model/Rank.java

```diff
@@ -3,38 +3,43 @@
 import java.util.stream.Stream;
 
 public enum Rank {
-    FIRST("1등", 6, 2_000_000_000),
-    SECOND("2등", 5, 1_500_000),
-    THIRD("3등", 4, 50_000),
-    FOURTH("4등", 3, 5_000),
-    NOTHING("꽝", 0, 0);
-
-    private final String name;
-    private final int matchedCount;
-    private final long prize;
-
-    Rank(String name, int matchedCount, int prize) {
-        this.name = name;
-        this.matchedCount = matchedCount;
-        this.prize = prize;
+    FIRST(6, 2_000_000_000),
+    SECOND(5, 30_000_000),
+    THIRD(5, 1_500_000),
+    FOURTH(4, 50_000),
+    FIFTH(3, 5_000),
+    MISS(0, 0);
+
+    private final int countOfMatch;
+    private final int winningMoney;
+
+    private Rank(int countOfMatch, int winningMoney) {
+        this.countOfMatch = countOfMatch;
+        this.winningMoney = winningMoney;
     }
 
-    public static Rank getRank(Long matchedCount) {
-        return getRank(matchedCount.intValue());
+    public int getCountOfMatch() {
+        return countOfMatch;
     }
 
-    public static Rank getRank(int matchedCount) {
-        return Stream.of(Rank.values())
-                .filter(rank -> rank.matchedCount == matchedCount)
-                .findFirst()
-                .orElse(Rank.NOTHING);
+    public int getWinningMoney() {
+        return winningMoney;
     }
 
-    public String getName() {
-        return name;
+    public static Rank valueOf(Long countOfMatch, boolean matchBonus) {
+        return valueOf(countOfMatch.intValue(), matchBonus);
     }
 
-    public long getPrize() {
-        return prize;
+    public static Rank valueOf(int countOfMatch, boolean matchBonus) {
+        Rank rank = Stream.of(Rank.values())
+                .filter(value -> value.countOfMatch == countOfMatch)
+                .findFirst()
+                .orElse(Rank.MISS);
+
+        if (rank == Rank.SECOND && !matchBonus) {
+            return Rank.THIRD;
+        }
+
+        return rank;
     }
 }
```

Comment:
```suggestion
}

```

파일에 EOL(End Of Line)을 추가해주세요~

* 여러 버전의 유닉스 OS간의 공통된 API 및 인터페이스를 정의해놓은 POSIX 명세 중 text file 부분을 보면 유닉스 시스템은 text file들을 서로 구분할때 EOL을 사용하기 때문에 파일을 만들때마다 파일의 끝에 EOL을 넣어주는 것이 안전하다.
* 구현하는 방법은 항상 파일 작성이 끝나면 엔터키를 통해 맨 아래줄에 공백을 만들어주거나, 프리티어를 통해 저장시 자동으로 파일 맨 끝에 개행이 되게끔 하는 방법이 있다.

Reply:
몰랐던 내용이네요ㅜ 피드백 감사합니다~